src/test/java/api/support/APITests.java

```diff
@@ -299,6 +299,7 @@ public void beforeEach() throws InterruptedException {
     usersFixture.defaultAdmin();
 
     FakePubSub.clearPublishedEvents();
+    FakePubSub.setFailPublishingWithNoSubscribersError(false);
```

Comment:
Is this ever enabled?

Reply:
Yes, in `CheckOutByBarcodeTests.checkOutSucceedsWhenEventPublishingFailsWithNoSubscribersError()`

Reply:
I think I misread the meaning of the boolean. 

Why do we want to turn this error off during the tests (given this is what the real implementation does)? Could that mask other failures, where messages are published?

Reply:
When this is enabled, the publishing will **always** fail with 400 Bad Request - it'll break other tests where we do need a successful publish. It's turned on just for one test to make sure we're handling "no subscribers" response correctly.

Reply:
@alexanderkurash Ok, thanks for explaining that to me.

> it'll break other tests where we do need a successful publish.

How is this observable in the tests? I thought this error was swallowed, or does it stop the fake pub-sub recording the messages?

Reply:
You're right, it's recorded in both cases. I'll change that so it's only stored in case of a successful publishing.

Reply:
Done